webui/module/Director/src/Director/Controller/DirectorController.php

```diff
@@ -90,6 +90,108 @@ public function indexAction()
     ));
   }
 
+  public function subscriptionAction()
+  {
+    $this->RequestURIPlugin()->setRequestURI();
+
+    if(!$this->SessionTimeoutPlugin()->isValid()) {
+      return $this->redirect()->toRoute(
+        'auth',
+        array(
+          'action' => 'login'
+        ),
+        array(
+          'query' => array(
+            'req' => $this->RequestURIPlugin()->getRequestURI(),
+            'dird' => $_SESSION['bareos']['director']
+          )
+        )
+      );
+    }
+
+    $module_config = $this->getServiceLocator()->get('ModuleManager')->getModule('Application')->getConfig();
+    $invalid_commands = $this->CommandACLPlugin()->getInvalidCommands(
+      $module_config['console_commands']['Director']['optional']
+    );
+    if(count($invalid_commands) > 0) {
+      $this->acl_alert = true;
+      return new ViewModel(
+        array(
+          'acl_alert' => $this->acl_alert,
+          'invalid_commands' => implode(",", $invalid_commands)
+        )
+      );
+    }
+
+    try {
+      $this->bsock = $this->getServiceLocator()->get('director');
+      $result = $this->getDirectorModel()->getStatusSubscriptionDetail($this->bsock);
+      $this->bsock->disconnect();
+    }
+    catch(Exception $e) {
+      echo $e->getMessage();
+    }
+
+    return new ViewModel(array(
+      'directorOutput' => $result
+    ));
+  }
+
+  public function backupunitreportAction()
```

Comment:
The functions subscriptionAction and backupunitreportAction are nearly identical. Can you please modify it to reuse the common code?

Reply:
Controller actions are already reduced to a limit to what needs to be there to deliver a new view model. Maybe we could argue on a refactor of the plugins called to get rid of conditions, but that should be done in a separate PR, because it affects all controllers and modules. It would be a bigger change which should not be part of this PR. If we really would like to touch this, we should create a story for that to optimize controller actions and plugin code, I think.